### PR TITLE
Build/deploy

### PR DESCRIPTION
**Pull request by tomrf1**
Now builds with teamcity, deploys with riffraff

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -10,18 +12,21 @@ val akkaVersion = "2.4.10"
 
 resolvers += Resolver.bintrayRepo("hseeberger", "maven")
 
+lazy val bot = (project in file(".")).enablePlugins(RiffRaffArtifact, JavaAppPackaging)
```
Comment (LATaylor-guardian) [inline build.sbt]: You can simply do: `enablePlugins(RiffRaffArtifact, JavaAppPackaging)` here.

File: cloudformation/cloudformation.yml [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+        Type: List<AWS::EC2::Subnet::Id>
+    LoggingKinesisStream:
+        Description: The name of the logging kinesis stream
+        Type: String
+    DynamodbTableName:
+        Description: Name of dynamodb table used to store user data
+        Type: String
+
+Mappings:
+    StageMap:
```
Comment (LATaylor-guardian) [inline cloudformation/cloudformation.yml]: Is this necessary? Your values are the same in both cases.

File: cloudformation/cloudformation.yml [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+        Properties:
+            PolicyName: BackendDeployPolicy
+            PolicyDocument:
+                Statement:
+                    - Effect: Allow
+                      Action:
+                        - s3:GetObject
+                      Resource:
+                        !Sub arn:aws:s3:::${DeployBucket}/*
+            Roles:
```
Comment (LATaylor-guardian) [inline cloudformation/cloudformation.yml]: This can be written as: `Roles: !Ref FacebookNewsBotRole`

File: cloudformation/cloudformation.yml [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+            PolicyDocument:
+                Statement:
+                    Effect: Allow
+                    Action:
+                        - ec2:DescribeInstances
+                        - ec2:DescribeTags
+                    Resource:
+                        - "*"
+            Roles:
+                - Ref: FacebookNewsBotRole
```
Comment (LATaylor-guardian) [inline cloudformation/cloudformation.yml]: As above.

File: cloudformation/cloudformation.yml [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+                    Effect: Allow
+                    Action:
+                        - dynamodb:GetItem
+                        - dynamodb:PutItem
+                        - dynamodb:UpdateItem
+                        - dynamodb:Query
+                    Resource:
+                        !Sub arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${DynamodbTableName}*
+            Roles:
+                - Ref: FacebookNewsBotRole
```
Comment (LATaylor-guardian) [inline cloudformation/cloudformation.yml]: As above.

File: cloudformation/cloudformation.yml [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+            PolicyName: dynamo-list-config-policy
+            PolicyDocument:
+                Statement:
+                    Effect: Allow
+                    Action:
+                        - dynamodb:ListTables
+                    Resource:
+                        - "*"
+            Roles:
+                - Ref: FacebookNewsBotRole
```
Comment (LATaylor-guardian) [inline cloudformation/cloudformation.yml]: As above.

File: cloudformation/cloudformation.yml [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+                          !Sub arn:aws:kinesis:${AWS::Region}:${AWS::AccountId}:stream/${LoggingKinesisStream}
+            Roles:
+                - Ref: FacebookNewsBotRole
+
+    InstanceProfile:
+        Type: AWS::IAM::InstanceProfile
+        Properties:
+            Path: /
+            Roles:
+                - Ref: FacebookNewsBotRole
```
Comment (LATaylor-guardian) [inline cloudformation/cloudformation.yml]: As above.

File: cloudformation/cloudformation.yml [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+
+    LoadBalancer:
+        Type: AWS::ElasticLoadBalancing::LoadBalancer
+        Properties:
+            Scheme: internet-facing
+            Subnets:
+                Ref: Subnets
+            CrossZone: true
+            SecurityGroups:
+                - Fn::GetAtt: [LoadBalancerSecurityGroup, GroupId]
```
Comment (LATaylor-guardian) [inline cloudformation/cloudformation.yml]: This can be written as `!GetAtt LoadBalancerSecurityGroup.GroupId`

File: facebook-news-bot.service [hunk 9]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=facebook-news-bot
+
+[Service]
+User=content-api
+Group=content-api
+Restart=on-failure
```
Comment (LATaylor-guardian) [inline facebook-news-bot.service]: This should be `Restart=no`. If for some reason the service can't start we want it to fail fast and default to the auto scaling group to take the instance out of service and replace with a new one.

File: project/build.properties [hunk 10]
```diff
@@ -0,0 +1 @@
+sbt.version=0.13.11
```
Comment (LATaylor-guardian) [inline project/build.properties]: 0.13.12 now.

File: cloudformation/cloudformation.yml (hunk 2 above)
Comment (tomrf1) [inline cloudformation/cloudformation.yml]: Not really, but one day we may need to scale up PROD...

**Comment by LATaylor-guardian** [comment]
👍

**Pull request merged**